docs/review.md

```diff
@@ -0,0 +1,579 @@
+  end
+end
+```
+
+* not so good
+
+```ruby
+scenario 'Current path is correct after signing up and peeping'  do
+  sign_up_and_create_peep('Test text')
+  expect(current_path).to eq '/peeps'
+end
+
+scenario 'Status code is correct after signing up and peeping'  do
+  sign_up_and_create_peep('Test text')
+  expect(page.status_code).to eq 200
+end
+
+scenario 'Peep is shown after signing up and peeping'  do
+  sign_up_and_create_peep('Test text')
+  within 'ul#peeps' do
+    expect(page).to have_content('Test text')
+  end
+end
+```
+
+
+### Correct location of feature/acceptance and unit tests
+
+Ensure that all your acceptance tests are in a separate folder called `features`.  This can be in your `spec` folder or on the root; up to you.
+
+If a test is in the feature folder it should be testing the entire stack, i.e. it should interact with the app via a web page, and then test the results of that action in the web page that gets returned.  It's acceptable to manipulate the database directly to set things up, but note their are risks associated here, i.e. that you will get your database in a state that it couldn't get into via the web interface, and thus when the tests pass it may not reflect precisely the user experience of using the site.
+
+Conversely if you are testing your models, with or without database interactions, then these tests should NOT be in your feature folder and should be in the `spec` folder, or in `spec/models`
+
+### Need unit tests for datamapper
+
+TODO I think we should encourage the use of a [shoulda](https://github.com/thoughtbot/shoulda-matchers) style syntax for datamapper:
+
+https://github.com/greyblake/dm-rspec
+
+But I realise that this might not be popular - but I think it's an important part of following the acceptance test, unit test cycle ...
+
+Maybe this should be part of moving to something like:
+
+* https://github.com/jeremyevans/sequel
+* https://github.com/kematzy/minitest-sequel
+* https://github.com/openhood/rspec_sequel_matchers
+
+I guess we can't have any of the above when the bookmark_manager project doesn't address any of this ...
+
+### Appropriate use of Spec Helpers
+
+```ruby
+require_relative 'helpers'
+
+feature "User sign up" do
+  include Helpers
+  let(:user) { user = build(:user) }
+
+  scenario 'with correct credentials' do
+    user = create(:user)
+    sign_in(user)
+    expect(page).to have_content("Welcome, #{user.email}")
+  end
+end
+```
+
+```ruby
+# spec/helpers.rb
+module Helpers
+
+  def sign_in(user)
+    visit('/sessions/new')
+    fill_in :email, with: user.email
+    fill_in :password, with: user.password
+    click_button "Log in"
+  end
+```
+
+See https://github.com/makersacademy/course/blob/master/pills/spec_helper_methods.md
+
+
+## Ensure asset routes are set correctly
+
+Do we like http://recipes.sinatrarb.com/p/asset_management/sinatra_assetpack ?
+
+Or should we just be telling them to set the public folder correctly
```

Comment:
Just set the public folder correctly for the moment, again we'd need to add this into the curriculum first otherwise